libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1124,6 +1124,62 @@ string YulUtilFunctions::clearStorageArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::clearStorageStructFunction(StructType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+
+	string functionName = "clear_struct_storage_" + _type.identifier();
+
+	return m_functionCollector.createFunction(functionName, [&] {
+		auto const& structType = dynamic_cast<StructType const&>(_type);
+		MemberList::MemberMap structMembers = structType.nativeMembers(nullptr);
+		vector<map<string, string>> memberSetValues;
+
+		set<u256> slotsCleared;
+		for (auto const& member: structMembers)
+			if (member.type->storageBytes() < 32)
```

Comment:
We could add a comment here, stating that we may want to check, whether, if we remove this case and handle everything like in the ``else``, the optimizer will eventually reduce that to just setting the whole slot to zero on its own.

Respectively, we could create an issue for investigating how well the optimizer deals with this right now and potentially tweaking it towards making the distinction here unnecessary.

The idea being that the storage loads, masking and write operations here won't be across loop iterations, but just happen sequentially, so in contrast to the array case, it might be feasible for the optimiser to deal with this.

But I think the current version should be safe and optimal without special optimization, so no need to stall this.

Reply:
I added comment here, so we have it as a reference.
I'll than open issue so we can track progress on optimizer. And will link this comment in issue, so we eventually change this.
Sounds good?

Reply:
Sounds good :-)!

Reply:
#9964